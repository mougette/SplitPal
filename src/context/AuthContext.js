import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
