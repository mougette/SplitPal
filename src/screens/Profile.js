import React, {useContext, useState, useEffect} from 'react';
import {View, Text, StyleSheet, Image } from 'react-native';
import {Button, Input, Icon} from 'react-native-elements';
import {Context as AuthContext} from '../context/AuthContext';
import SplitPalLogoComponent from '../components/SplitPalLogoComponent';
import {Get} from '../components/RestGet';
import {Put} from '../components/RestPut';

const Profile = ({navigation}) => {
  const {state, signout} = useContext(AuthContext);
  const [disabled, setDisabled] = useState(true);
  const [changeButton, setChangeButton] = useState("Change Profile");
  const [newFirstName, setNewFirstName] = useState('NULL');
  const [newLastName, setNewLastName] = useState('NULL');
  const [newPhone, setNewPhone] = useState('1234567890');
  const [DATA, setDATA] = useState("");

  useEffect(() => {
    const unsubscribe = navigation.addListener('focus', () => {
      Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/profile","?email="+state.email)
        .then(response => {
          let resp = JSON.parse(response)
          console.log(resp)
          setNewFirstName(resp[0].firstName)
          setNewLastName(resp[0].lastName)
          setNewPhone(resp[0].phoneNumber)
        });
      });
      return unsubscribe
    }, [navigation]);


  function handleEditClick() {
    setDisabled(!disabled);
    if (disabled == true) {
      setChangeButton("Save Profile")
    }
    else {
      setChangeButton("Change Profile")
      console.log("/PUT updating profile info")
      Put("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/profile",
                  JSON.stringify({
                   userEmail: state.email,
                   newEmail: state.email,
                   firstName: newFirstName,
                   lastName: newLastName,
                   phoneNumber: newPhone,
                   profilePic: "NULL",
                 }))
    }
  }

  return (
    <View style={styles.master}>
      <SplitPalLogoComponent />
      <Text style={styles.header}>Welcome, {state.email}</Text>
      <Image source ={{uri: 'https://reactnative.dev/img/tiny_logo.png'}}
             style={{width:60, height:60,borderRadius:30}} />
      <Input
         placeholder="NULL"
         onChangeText={setNewFirstName}
         value={newFirstName}
         label="First Name"
         leftIcon={<Icon name="github" type="font-awesome" size={24} />}
         autoCorrect={false}
         disabled={disabled}
       />
       <Input
          placeholder="NULL"
          onChangeText={setNewLastName}
          value={newLastName}
          label="Last Name"
          leftIcon={<Icon name="github" type="font-awesome" size={24} />}
          autoCorrect={false}
          disabled={disabled}
        />
        <Input
           placeholder="NULL"
           onChangeText={setNewPhone}
           value={newPhone}
           label="Phone Number"
           leftIcon={<Icon name="phone" type="font-awesome" size={24} />}
           autoCorrect={false}
           disabled={disabled}
         />
      <Button onPress={handleEditClick} title={changeButton} type="clear" />
      <Button onPress={() => navigation.navigate("Change Password")}
        title="Change Password" type="clear" />
      <Button onPress={signout} title="Ready to Sign out?" type="clear" />
    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    fontSize: 20,
    marginBottom: 8,
    marginTop: -20,
  },
});

export default Profile;
