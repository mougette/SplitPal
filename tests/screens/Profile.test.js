import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Profile from '../../src/screens/Profile'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("[{\"firstName\": \"NULL\", \"lastName\": \"NULL\", \"email\": \"e@something.com\", \"phoneNumber\": \"NULL\", \"profilePicture\": null}]"),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Profile tests', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Profile', async () => {
        const {getByText} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="Profile" component={Profile} />
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        await waitFor(() => {
        fireEvent.press(getByText("Change Profile"))
        })
        fireEvent.press(getByText("Save Profile"))
        expect(1).toBe(1);
        })
    it('Profile', async () => {
            const {getByText} = render(
            <AuthProvider>
            <NavigationContainer>
            <Stack.Navigator>
            <Stack.Screen name="Profile" component={Profile} />
            </Stack.Navigator>
            </NavigationContainer>
            </AuthProvider>);
            await waitFor(() => {
            fireEvent.press(getByText("Ready to Sign out?"))
            })
            expect(1).toBe(1);
            })
        });


