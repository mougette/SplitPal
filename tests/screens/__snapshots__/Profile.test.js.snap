// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests Profile 1`] = `undefined`;

exports[`Profile tests Profile 2`] = `undefined`;
